### PR TITLE
Introduced new committee supervisor role that allows the committee host to promote adherents as committee hosts.

### PR DESCRIPTION
**Pull request by hhamon**

### CODE REVIEW

File: src/AppBundle/Committee/CommitteeManager.php [hunk 1]
```diff
@@ -140,6 +142,23 @@ public function getCommitteeMembers(Committee $committee): AdherentCollection
     }
 
     /**
+     * Promotes an adherent to be a host of a committee.
+     *
+     * @param Adherent  $adherent
+     * @param Committee $committee
+     * @param bool      $flush
+     */
+    public function promote(Adherent $adherent, Committee $committee, bool $flush = true)
```
Comment (mykiwi) [inline src/AppBundle/Committee/CommitteeManager.php]: `: void` ?

File: src/AppBundle/Committee/Voter/SuperviseCommitteeVoter.php [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+class SuperviseCommitteeVoter extends Voter
+{
+    private $committeeMembershipRepository;
+
+    public function __construct(CommitteeMembershipRepository $committeeMembershipRepository)
+    {
+        $this->committeeMembershipRepository = $committeeMembershipRepository;
+    }
+
+    protected function supports($attribute, $subject)
```
Comment (mykiwi) [inline src/AppBundle/Committee/Voter/SuperviseCommitteeVoter.php]: `: bool`

File: src/AppBundle/Committee/Voter/SuperviseCommitteeVoter.php [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    }
+
+    /**
+     * @param string         $attribute
+     * @param Committee      $committee
+     * @param TokenInterface $token
+     *
+     * @return bool
+     */
+    protected function voteOnAttribute($attribute, $committee, TokenInterface $token)
```
Comment (mykiwi) [inline src/AppBundle/Committee/Voter/SuperviseCommitteeVoter.php]: `: bool`

File: src/AppBundle/Entity/Adherent.php [hunk 4]
```diff
@@ -426,6 +426,18 @@ public function updateMembership(MembershipRequest $membership, PostAddress $pos
     }
 
     /**
+     * Joins a committee as a SUPERVISOR priviledged person.
+     *
+     * @param Committee $committee
+     *
+     * @return CommitteeMembership
+     */
+    public function superviseCommittee(Committee $committee)
```
Comment (mykiwi) [inline src/AppBundle/Entity/Adherent.php]: return type hint missing

File: src/AppBundle/Entity/CommitteeMembership.php [hunk 5]
```diff
@@ -194,4 +239,13 @@ public function setPrivilege(string $privilege)
     {
         $this->privilege = $privilege;
     }
+
+    public function promote()
```
Comment (mykiwi) [inline src/AppBundle/Entity/CommitteeMembership.php]: `: void`

File: src/AppBundle/Committee/Voter/SuperviseCommitteeVoter.php (hunk 2 above)
Comment (hhamon) [inline src/AppBundle/Committee/Voter/SuperviseCommitteeVoter.php]: it's from Symfony API so I don't know if we can change the signature.

File: src/AppBundle/Entity/Adherent.php (hunk 4 above)
Comment (hhamon) [inline src/AppBundle/Entity/Adherent.php]: Good catch.

File: src/AppBundle/Committee/Voter/SuperviseCommitteeVoter.php (hunk 2 above)
Comment (hhamon) [inline src/AppBundle/Committee/Voter/SuperviseCommitteeVoter.php]: Looks like we can do it.

**Comment by hhamon** [comment]
Addressed comments.

**Comment by ngrilly** [comment]
This is great! When do you expect to deploy this in the production environment?

**Comment by hhamon** [comment]
@ngrilly it should be deployed very soonly I guess. @tgalopin can tell you when exactly. It must be tested on a preproduction server first.

**Comment by tgalopin** [comment]
@ngrilly this is experimental for now (it needs front-end work) but should be deployed quite soon indeed.

**Comment by ngrilly** [comment]
Ok, thanks a lot.

**Pull request merged**